UI/window-basic-main-profiles.cpp

```diff
@@ -813,6 +823,16 @@ void OBSBasic::ChangeProfile()
 			close();
 		}
 	}
+
+#ifdef YOUTUBE_ENABLED
+	if (YouTubeAppDock::IsYTServiceSelected()) {
+		// This is re-initialized in some cases.
+		youtubeAppDock = main->GetYouTubeAppDock();
+		if (youtubeAppDock != NULL) {
+			youtubeAppDock->show();
+		}
+	}
```

Comment:
Same as above.
```suggestion
	if (YouTubeAppDock::IsYTServiceSelected() && youtubeAppDock)
		youtubeAppDock->show();
```

Reply:
I can make all these changes except for line 830. In the case where the user is logged in via a YouTube account (vs using streamkey), the reference to the dock changes and you need to recall main->GetYouTubeAppDock();

WDYT?

Reply:
You are in `OBSBasic` here so `main->GetYouTubeAppDock()` and `youtubeAppDock` will point to the same object.

Reply:
Shared repro steps here: https://youtu.be/Z23FcPHwUGA

Only crashes on YouTube connected profiles.

Reply:
You need to apply both my suggested changes, otherwise you're still shadowing the member variable with the one declared in the first part of the diff, which may no longer point to a valid object.